test/unit/RedisResourceManagerTest.php

```diff
@@ -71,6 +72,27 @@ public function testSetServerWithPasswordInParameters(): void
         $this->assertEquals('testhost', $server['host']);
         $this->assertEquals(1234, $server['port']);
         $this->assertEquals('abcd1234', $this->resourceManager->getPassword($dummyResId2));
+        $this->assertEquals('dummyuser', $this->resourceManager->getUser($dummyResId2));
+    }
+
+    public function testSetServerWithPasswordInParametersAndNoUser(): void
```

Comment:
can you add a test to verify that extracting the user from a URI works as expected?

You introduced a way to extract the username from `redis://username:password@testhost:1234` but I do not see a test for that.